src/main/java/org/opensearch/flowframework/util/EncryptorUtils.java

```diff
@@ -227,7 +228,15 @@ public Template redactTemplateCredentials(Template template) {
             processedWorkflows.put(entry.getKey(), new Workflow(entry.getValue().userParams(), processedNodes, entry.getValue().edges()));
         }
 
-        return new Template.Builder(template).workflows(processedWorkflows).build();
+        Template updatedTemplate;
+
+        if (template.getUser() != null) {
+            updatedTemplate = new Template.Builder(template).name(template.name()).user(null).build();
+        } else {
+            updatedTemplate = template;
+        }
+
+        return new Template.Builder(updatedTemplate).workflows(processedWorkflows).build();
```

Comment:
```suggestion
        return new Template.Builder(template).user(null),workflows(processedWorkflows).build();
```

Reply:
Wouldn't this add a user field in the builder, although null but even when it's not present?

Reply:
There already is a user field initialized thus:
```java
private User user = null;
```
So no difference here with an uninitialized vs. intentionally cleared.

Reply:
Ah! Sure, that makes sense. I like the one line change now for this method.